cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1662,7 +1662,348 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
 
         Opcode::Imul => {
             let ty = ty.unwrap();
-            if ty == types::I64X2 {
+
+            // First check for ext_mul_* instructions. Where possible ext_mul_* lowerings
```

Comment:
I think the order of the lowerings should be reversed: scalar lowerings first, then vector lowerings, then pattern matching lowerings. As it stands in this PR, every `imul.i64` is going to require a bunch of matching calls and type comparisons before it ever gets lowered. I think scalar multiplication is going to be the most common, then plain vector multiplication, so I would think we would want to make those common paths shorter. (Not sure exactly how much this affects compile time but hopefully you see what I mean).

Reply:
Sure .. I agree. That was my original thought but I never reverted the initial implementation.

Reply:
Also note, the code must remain at the top as the fall through target since I am using "if let Some(_) .." to check for op sources and there is no if not let Some(_) ..". Still there is just one branch to get to the other lowerings.